File: algorithms/bit/find_xor_sum_of_all_pairs_bitwise_and.rs
// https://leetcode.com/problems/find-xor-sum-of-all-pairs-bitwise-and/

// time complexity: O(n)

impl Solution {
    pub fn get_xor_sum(arr1: Vec<i32>, arr2: Vec<i32>) -> i32 {
        let (mut a, mut b) = (0, 0);
        for n in &arr1 {
            a ^= n;
        }
        for n in &arr2 {
            b ^= n;
        }
        return a & b;
    }
}

File: leetcode/find_xor_sum_of_all_pairs_bitwise_and.cc
// https://leetcode.com/problems/find-xor-sum-of-all-pairs-bitwise-and/

// time complexity: O(n)

#include <vector>

using namespace std;

class Solution {
public:
    int getXORSum(vector<int> &arr1, vector<int> &arr2) {
        int a = 0, b = 0;
        for (const auto &n : arr1)
            a ^= n;
        for (const auto &n : arr2)
            b ^= n;
        return a & b;
    }
};